horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/RunServiceImpl.java

```diff
@@ -1004,4 +1012,213 @@ public DatasetList listDatasetsBySchema(String uri, Integer limit, Integer page,
+                     }
+
+                     extractorObjectsPerTrans.put(t, root);
+                  }
+
+                  boolean containsArray = false;
+                  int arrayPosition = -1;
+
+                  for (Transformer t : transformers) {
+                     ObjectNode objects = extractorObjectsPerTrans.get(t);
+                     if (t.function == null || t.function.isBlank()) {
+                        dataSets.add(new DataSet(run.start, run.stop
+                           , "", test.id
+                           , objects, run
+                           , ordinal++, run.owner, run.access));
+                     } else {
+                        StringBuilder code = new StringBuilder();
+                        String obj = objects.toString();
+                        code.append("const __obj = ").append(obj).append(" ;\n");
+                        code.append("const __func = ").append(t.function).append(";\n");
+                        code.append("__func(__obj);\n");
+                        String result = null;
+                        ByteArrayOutputStream out = new ByteArrayOutputStream();
+                        try (org.graalvm.polyglot.Context context = org.graalvm.polyglot.Context.newBuilder("js").out(out).err(out).build()) {
+                           context.enter();
+                           try {
+                              Value value = context.eval("js", code.toString());
+                              result = value.toString();
+                           } catch (PolyglotException e) {
+                              log.errorf("Failed to execute transformation for run %d and test %d and cause %s", run.id, test.id, e);
+                              if (result != null) {
+                                 logTransformationMessage(test, run, TransformationLog.ERROR, "transformation "+t.name+" function", result);
+                                 functionError = true;
+                              }
+                           } finally {
+                              context.leave();
+                              if (out.size() > 0) {
+                                 logTransformationMessage(test, run, TransformationLog.ERROR, "transformation "+t.name+" function", result);
+                              }
+                           }
+                        }
+                        if (result != null) {
+                           JsonNode node = Util.toJsonNode(result);
+                           if (node == null || functionError) {
+                              log.errorf("Conversion of Transformation "+t.name+" function result failed. See prior logs for cause.");
+                              node = instance.objectNode();
+                              transformedSingleNodeResults.add(node);
+                           } else if (node.isArray() ) {
+                              // let's assume (always dangerous) only 1 Transformer produces an array
```

Comment:
This assumption is too bold, rather make it `List<List<JsonNode>> transformedArrayNodeResults`.

Reply:
When the sizes of 2 arrays means the remainder is non zero what do you wish to happen ?
extractor sizes: 1, 3 and 8